src/Marketplace/Services/RecipientService.php

```diff
@@ -86,4 +88,22 @@ public function saveRecipient(Recipient $recipient)
 
         return $recipient;
     }
+
+    /**
+     * @param $sellerId
+     * @throws CouldNotSaveException
+     */
+    public function findRecipient($sellerId)
+    {
+        $recipient = $this->recipientRepository->findBySellerId($sellerId);
+
+        if(empty($recipient)) {
+            $this->logService->info(
+                __("The seller does not have a registered recipient.")
+            );
+            throw new CouldNotSaveException(
```

Comment:
A gente não pode lançar essa exception aqui porque ela é do Magento e o Core não pode fazer uso de recursos exclusivos de plataformas.

Se a gente futuramente for implementar o split no Woocommerce, essa linha vai quebrar o core porque lá não vai ter o path `Magento\Framework\Exception\CouldNotSaveException`.

Nesse caso a gente precisaria lançar uma exception do core mesmo, e verificar como podemos capturar essa exception no código do módulo magento 2 pra lançar a exception `CouldNotSaveException`, faz sentido?

Reply:
Faz sentido! corrigido!